tonic/models/vic/grid_params.py

```diff
@@ -917,6 +1150,34 @@ def grid_params(soil_dict, target_grid, snow_dict, veglib_dict, veg_dict,
                 for y, x in pyzip(ymask, xmask):
                     out_dict[var][:, :, y, x] = fill_val
 
+            elif mydict[var].ndim == 4:
+                j = mydict[var].shape[1]
+                k = mydict[var].shape[2]
+                l = mydict[var].shape[3]
+                out_dict[var] = np.ma.zeros((j, k, l, ysize, xsize),
+                                            dtype=dtype)
+                for jj in pyrange(j):
+                    for kk in pyrange(k):
+                        for ll in pyrange(l):
+                            out_dict[var][jj, kk, ll, yi, xi] = mydict[var][:, jj, kk, ll]
+                for y, x in pyzip(ymask, xmask):
+                    out_dict[var][:, :, :, y, x] = fill_val
+
+            elif mydict[var].ndim == 5:
+                j = mydict[var].shape[1]
+                k = mydict[var].shape[2]
+                l = mydict[var].shape[3]
+                m = mydict[var].shape[4]
+                out_dict[var] = np.ma.zeros((j, k, l, m, ysize, xsize),
+                                            dtype=dtype)
+                for jj in pyrange(j):
+                    for kk in pyrange(k):
+                        for ll in pyrange(l):
+                            for mm in pyrange(m):
+                                out_dict[var][jj, kk, ll, mm, yi, xi] = mydict[var][:, jj, kk, ll, mm]
+                for y, x in pyzip(ymask, xmask):
+                    out_dict[var][:, :, :, :, y, x] = fill_val
+
             out_dict[var] = np.ma.masked_values(out_dict[var], fill_val)
```

Comment:
The reason I didn't use `xarray` was that it didn't exist when I wrote this code. I'd like to refactor this while module to use xarray but that will be quite an undertaking and should not be done by @tbohn in this PR.